internal/bucket/access.go

```diff
@@ -37,13 +38,40 @@ func (b *bucketImpl) GetContent() []*BucketContent {
 	return content
 }
 
-func (b *bucketImpl) GetBuckets() []*BucketContent {
-	var content []*BucketContent = make([]*BucketContent, 0)
-	for _, v := range b.contentCache {
-		if v.ContentType == bt.ResourceIdentifier_DID {
-			content = append(content, v)
-		}
+func (b *bucketImpl) GetBuckets() []Bucket {
+	var content []Bucket = make([]Bucket, 0)
+	for _, v := range b.bucketCache {
+		content = append(content, v)
 	}
 
 	return content
 }
+
+func (b *bucketImpl) ResolveContentBySchema(did string) ([]*mt.BucketContent, error) {
+	var matchedContent []*mt.BucketContent = make([]*mt.BucketContent, 0)
+	for _, c := range b.whereIs.Content {
+		if c.Type == bt.ResourceIdentifier_CID {
+			if c.SchemaDefinition.Did == did {
```

Comment:
If you're only checking resources with CIDs, won't none of the DID fields be present?

Reply:
the identifier check is to make sure that the content is not a bucket, then the check on the given schema did is to check if the schema being queried from outside the scope is referenced by content within the specified bucket. `SchemaReference` is now just a `DID` as the entire reference was not needed